superset/connectors/sqla/models.py

```diff
@@ -347,7 +347,7 @@ class SqlaTable(Model, BaseDatasource):
     owner_class = security_manager.user_model
 
     __tablename__ = "tables"
-    __table_args__ = (UniqueConstraint("database_id", "table_name"),)
+    __table_args__ = (UniqueConstraint("database_id", "schema", "table_name", name="_customer_location_uc"),)
```

Comment:
@john-bodley

Reply:
@willbarett this wont work per https://github.com/apache/incubator-superset/pull/5449.

Reply:
@john-bodley this change is on the model to bring it in line with the existing constraint at the database layer. Should we instead be adding a migration to remove the existing table constraint? How would you recommend we reconcile the model definitions with the table schema in this instance?

Reply:
The model needs to reflect the table schema. As far as I can tell the only viable solution is an approach similar to #5449.

Reply:
Right. This is altering the model to match the table schema. Can you expand on what you see the problem to be? I'm not following what you mean from reading the referenced PR.

Reply:
I don't believe the table schema has a uniqueness constraint on the (`database_id`, `schema`, `table_name`) tuple as this violates the MySQL constraint length. Additionally since schema can be `NULL` depending on how the underlying engine (MySQL, Postgres, etc.) deals with `NULL` values for uniqueness the constraint _may_ actually allow for duplicate records. 

#5449 proposes a solution which is entirely handled by SQLAlchemy.

Reply:
I connected with Max - it appears there's a bug in the migration that results in the old constraint not being dropped in some circumstances. I'm looking into that. The intention here isn't to alter the existing constraint on this table. Will update.